### PR TITLE
Reduce jsb glue code size

### PR DESCRIPTION
**Pull request by PatriceJiang**
Re: #

### Changelog

* Un-inline jsb wrapper functions, reduce the size of libcocos.so by around 1MB

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/bindings/jswrapper/State.cpp [hunk 1]
```diff
@@ -27,24 +27,4 @@
...
-State::State(Object *thisObject, const ValueArray &args)
-: _thisObject(thisObject),
-  _args(&args) {
-    if (_thisObject != nullptr) {
-        _thisObject->incRef();
-    }
-}
-
-State::~State() {
-    SAFE_DEC_REF(_thisObject);
```
Comment (dumganhar) [inline native/cocos/bindings/jswrapper/State.cpp:47]: Moving the code to header could make compiled code smaller?

Comment (PatriceJiang) [reply native/cocos/bindings/jswrapper/State.cpp:47]: To inline the destructor and code size is bearly not aaffected.

**Review by dumganhar — APPROVED** [review]

**Comment by PatriceJiang** [comment]
@cocos-robot run test cases

**Comment by PatriceJiang** [comment]
@cocos-robot run test cases

File: native/cocos/bindings/manual/jsb_cocos_manual.cpp [hunk 2]
```diff
@@ -36,7 +36,7 @@
 
 extern se::Object *__jsb_cc_FileUtils_proto; // NOLINT(readability-redundant-declaration, readability-identifier-naming)
 
-static bool jsb_ccx_empty_func(const se::State & /*s*/) { // NOLINT(readability-identifier-naming)
+static bool jsb_ccx_empty_func(se::State & /*s*/) { // NOLINT(readability-identifier-naming)
```
Comment (minggo) [inline native/cocos/bindings/manual/jsb_cocos_manual.cpp:39]: Why remove const?

File: native/cocos/bindings/jswrapper/State.cpp [hunk 3]
```diff
@@ -27,24 +27,4 @@
...
-}
```
Comment (minggo) [inline native/cocos/bindings/jswrapper/State.cpp:48]: Any reason to implement it in .h?

File: native/cocos/bindings/manual/jsb_cocos_manual.cpp (hunk 2 above)
Comment (PatriceJiang) [reply native/cocos/bindings/manual/jsb_cocos_manual.cpp:39]: All jsb function should be unified and meet the signature `using se_function_ptr = bool (*)(se::State &state);` as an parameter.

File: native/cocos/bindings/jswrapper/State.cpp (hunk 3 above)
Comment (PatriceJiang) [reply native/cocos/bindings/jswrapper/State.cpp:48]: The code size of this function is somewhats small and `se::State` objects are only constructed and destructed within `jsb*Wrapper` functions, which are not frequently used.  By inlining the destructor fewer function calls will be generated within JSB functions.

Comment (minggo) [reply native/cocos/bindings/jswrapper/State.cpp:48]: Yep, the code size is small. But if there is not special reason, then should implement in .cp by default.

File: native/cocos/bindings/manual/jsb_cocos_manual.cpp (hunk 2 above)
Comment (minggo) [reply native/cocos/bindings/manual/jsb_cocos_manual.cpp:39]: ok

File: native/cocos/bindings/jswrapper/State.cpp (hunk 3 above)
Comment (PatriceJiang) [reply native/cocos/bindings/jswrapper/State.cpp:48]: Yes, for most cases destructors are implemented in source files. Here these `jsb*Wrapper` functions are hot and inlined.

Comment (minggo) [reply native/cocos/bindings/jswrapper/State.cpp:48]: Ok

Comment (minggo) [reply native/cocos/bindings/jswrapper/State.cpp:48]: Please do testing about it. We should do decision base on testing result. If it is true, then it is a new rule, we should check other source code too.

Comment (PatriceJiang) [reply native/cocos/bindings/jswrapper/State.cpp:48]: ### Code size
||final code size|
|--|----|
|inline| 21365912 B (+208B)|
|non-inline|21365704 B|

### Instructions

### non-inline
image
image

#### inline
image

less instructions are emitted, and should be a bit faster.

**Review by minggo — APPROVED** [review]

**Pull request merged**